py/desitarget/lyazcat.py

```diff
@@ -0,0 +1,441 @@
+                                'Z_COMB_PROB','Z_QN','Z_QN_CONF','IS_QSO_QN',
+                                'Z_SQ','Z_SQ_CONF','Z_ABS','Z_ABS_CONF'])
+"""
+zcatdatamodel_formats = np.array(['>int64','>float64','>int64','>float64',
+                                  '>float64','>float64','>float64','>float64',
+                                  '>int16','>float64','>float64','>float64',
+                                  '>float64'])
+"""
+zcatdatamodel_formats = np.array(['>i8','>f8','>i8','>f8',
+                                  '>f8','>f8','>f8','>f8',
+                                  '>i2','>f8','>f8','>f8',
+                                  '>f8'])
+zcols_copy = np.array(['TARGETID','Z','ZWARN','DELTACHI2'])
+n1_cols = np.array(['Z_QN','Z_QN_CONF','IS_QSO_QN',
+                    'Z_SQ','Z_SQ_CONF','Z_ABS','Z_ABS_CONF'])
+
+
+def tmark(istring):
+    """A function to mark the time an operation starts or ends.
+    
+    Parameters
+    ----------
+    istring : :class:'str'
+        The input string to print to the terminal.
+        
+    Output
+    ------
+    :class:'str'
+        A string with the date and time in ISO 8061 standard followed
+        by the istring.
+    """
+    t0 = time.time()
+    t_start = time.strftime('%Y-%m-%d | %H:%M:%S')
+    print('\n{}: {}'.format(istring,t_start))
+
+    
+def make_new_zcat(zbestname):
+    """Make the initial zcat array with redrock data.
+    
+    Parameters
+    ----------
+    zbestname : :class:`str`
+        Full filename and path for the zbest file to process.
+        
+    Returns
+    -------
+    :class:`~numpy.array`
+        A zcat in the official format (`zcatdatamodel`) compiled from
+        the `tile', 'night', and 'petal_num', in `zcatdir`.
+    """
+    tmark('    Making redrock zcat')
+    try:
+        zs = fits.open(zbestname)["ZBEST"].data
+        # EBL write out the zcat as a file with the correct data model.
+        zcat = np.zeros(len(zs), dtype={'names': zcatdatamodel_names, 'formats': zcatdatamodel_formats})
+        for col in zcols_copy:
+            zcat[col] = zs[col]
+        for col in n1_cols:
+            zcat[col][:] = -1
+
+        return zcat
+    except (FileNotFoundError, OSError):
+        return False
+
+    
+def add_qn_data(zcat, coaddname):
+    """Apply the QuasarNP model to the input zcat and add data to columns.
+    
+    Parameters
+    ----------
+    zcat : :class:'~numpy.array'
+        The structured array that was created by make_new_zcat()
+    coaddname : class:'str'
+        The name of the coadd file corresponding to the zbest file used
+        in make_new_zcat()
+        
+    Returns
+    -------
+    :class:'~numpy.array'
+        The zcat array with QuasarNP data included in the columns:
+        * Z_QN
+        * Z_QN_CONF
+        * IS_QSO_QN
+    """
+    tmark('    Adding QuasarNP data')    
+    lines = ['LYA','CIV(1548)','CIII(1909)', 'MgII(2796)','Hbeta','Halpha']
```

Comment:
Code style. Need a white-space after every comma in a list.

Reply:
Fixed for the next push.